resolvers/node/resolution.rs

```diff
@@ -493,17 +498,36 @@ impl<
     let maybe_file_type = self.sys.get_file_type(&path);
     match maybe_file_type {
       Ok(FileType::Dir) => {
-        let suggested_file_name = ["index.mjs", "index.js", "index.cjs"]
+        let ext = ["js", "mjs", "cjs"]
           .into_iter()
-          .find(|e| self.sys.is_file(&path.join(e)));
-        Err(
-          UnsupportedDirImportError {
-            dir_url: UrlOrPath::Path(path),
-            maybe_referrer: maybe_referrer.map(|r| r.display()),
-            suggested_file_name,
-          }
-          .into(),
-        )
+          .find(|ext| self.sys.is_file(&with_known_extension(&path, ext)));
+        let suggested_file_name =
+          if resolution_mode == ResolutionMode::Import || ext.is_none() {
+            ["index.mjs", "index.js", "index.cjs"]
+              .into_iter()
+              .find(|e| self.sys.is_file(&path.join(e)))
+          } else {
+            None
+          };
+        if resolution_mode == ResolutionMode::Import
+          || (ext.is_none() && suggested_file_name.is_none())
+        {
+          let suggested_file_name = ["index.mjs", "index.js", "index.cjs"]
```

Comment:
Node doesn't resolve index.mjs/cjs files and doesn't automatically resolve .cjs and .mjs extensions. I want this code right now so I'm going to extract this code out from this PR and make a few fixes as well as add some tests.